include/arch/arm/aarch32/cortex_m/scripts/linker.ld

```diff
@@ -379,8 +380,55 @@ SECTIONS
 #include <linker/priv_stacks-noinit.ld>
 #include <linker/cplusplus-ram.ld>
 
-    __data_ram_end = .;
+__data_ram_end = .;
+
+#ifndef CONFIG_USERSPACE
+   SECTION_DATA_PROLOGUE(_BSS_SECTION_NAME,(NOLOAD),)
+	{
+        /*
+         * For performance, BSS section is assumed to be 4 byte aligned and
+         * a multiple of 4 bytes
+         */
+        . = ALIGN(4);
+	__bss_start = .;
+	__kernel_ram_start = .;
+
+	*(.bss)
+	*(".bss.*")
+	*(COMMON)
+	*(".kernel_bss.*")
+
+#ifdef CONFIG_CODE_DATA_RELOCATION
+#include <linker_sram_bss_relocate.ld>
+#endif
 
+        /*
+         * As memory is cleared in words only, it is simpler to ensure the BSS
+         * section ends on a 4 byte boundary. This wastes a maximum of 3 bytes.
+		 */
+	__bss_end = ALIGN(4);
+	} GROUP_DATA_LINK_IN(RAMABLE_REGION, RAMABLE_REGION)
+
+    SECTION_PROLOGUE(_NOINIT_SECTION_NAME,(NOLOAD),)
+        {
+        /*
+         * This section is used for non-initialized objects that
+         * will not be cleared during the boot process.
+         */
+        *(.noinit)
+        *(".noinit.*")
+	*(".kernel_noinit.*")
+
+/* Located in generated directory. This file is populated by the
+ * zephyr_linker_sources() Cmake function.
+ */
+#include <snippets-noinit.ld>
+#ifdef CONFIG_SOC_NOINIT_LD
+#include <soc-noinit.ld>
+#endif
+
+        } GROUP_LINK_IN(RAMABLE_REGION)
+#endif /*CONFIG_USERSPACE*/
```

Comment:
To be consistent with others: `/* CONFIG_USERSPACE */`

Reply:
Can you give some suggestions, is there a formal way to distinguish the two `/ * CONFIG_USERSPACE * / `  Thanks

Reply:
I meant the space after and before `*`.

`/* CONFIG_USERSPACE */`
`/*CONFIG_USERSPACE*/`

Reply:
Also, although not strictly a rule, it is common to do add `!` for `ifndef`s, e.g. `/* !CONFIG_USERSPACE */` to distinguish from `ifdef`s.

Reply:
Thanks for your explain, I fixed it